### DISCUSSION TITLE
about using upx

### DISCUSSION THREAD

**Discussion by baosj0** [pyinstaller/pyinstaller#6132] [PyInstaller] [answered]
from the doc:
https://pyinstaller.readthedocs.io/en/stable/usage.html?highlight=upx#using-upx

 In the case of a PyInstaller one-file executable that has been UPX-compressed, the full execution sequence is:
The compressed program start up in the UPX decompressor code.
After decompression, the program executes the PyInstaller bootloader, which creates a temporary environment for Python.
The Python interpreter executes your script.

i'm using like this: pyinstaller --upx-dir=xxxx  -F  1.py
and generating a 1.exe,   but this exe is not upx-packed.  the binaries embedded such as .pyd, .dll are upx-packed..

why dont pyinstaller pack the final output binary automatically?

**Comment by bwoodsend** [comment, marked as answer]
Because there is no point compressing an already compressed file. It'll give negligible size improvement (possibly even making it bigger) whilst slowing down startup because it needs to decompress everything twice.

**Reply by baosj0** [reply]
so that document is wrong?

**Reply by rokm** [reply]
Yeah, the closing sentences of that documentation section are incorrect.

**Reply by baosj0** [reply]
thanks for replying

**Reply by rokm** [reply]
The offending piece of documentation has been revised in #6161.
